grin/src/Reducer/Interpreter/Base.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE LambdaCase, GeneralizedNewtypeDeriving, InstanceSigs, TypeFamilies, TemplateHaskell, ScopedTypeVariables, DataKinds #-}
+{-# LANGUAGE TypeOperators, DeriveFunctor, ViewPatterns, EmptyCase, RankNTypes #-}
+module Reducer.Interpreter.Base
+  ( module Reducer.Interpreter.Base
+  ) where
+
+import Control.Monad.Fail
+import Control.Monad.Trans (MonadIO)
+import Data.Function (fix)
+import Data.Map.Strict (Map(..), fromList)
+import Grin.ExtendedSyntax.Syntax hiding (Val)
+import Data.Functor.Foldable
+-- import Grin.Value hiding (Val)
+
+import Reducer.Interpreter.Env (Env)
+import qualified Reducer.Interpreter.Env as Env
+import qualified Grin.ExtendedSyntax.Syntax as Grin (Val)
+import Transformations.ExtendedSyntax.Util (anaM)
+
+
+-- * Interpreter
+
+data (:+:) f g a = Inl (f a) | Inr (g a)
```

Comment:
Hmm, now I see what this means. Did you consider using the definition in [`base`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Functor-Sum.html)?

Reply:
Ah finally the `base` also have this. Last time I checked it wasn't there. Maybe I do Haskell long enough to refresh my foundations? :)

Reply:
I really liked the infix version though :/ Maybe a pattern synonym? :D

Reply:
Not yet :)